File: monsoon3.0/src/main/java/wtf/monsoon/api/setting/Bind.java
package wtf.monsoon.api.setting;

import lombok.Getter;
import lombok.Setter;
import org.lwjgl.input.Keyboard;
import org.lwjgl.input.Mouse;

/**
 * @author Surge
 * @since 08/08/2022
 */
public class Bind {

    // The button index
    @Getter
    @Setter
    private int buttonCode;

    // Input device
    @Getter
    @Setter
    private Device device;

    public Bind(int buttonCode, Device device) {
        this.buttonCode = buttonCode;
        this.device = device;
    }

    public boolean isPressed() {
        if (buttonCode == 0) {
            return false;
        }

        // Our bind is pressed
        boolean pressed = Keyboard.isKeyDown(buttonCode) && device.equals(Device.KEYBOARD) || Mouse.isButtonDown(buttonCode) && device.equals(Device.MOUSE);

        return pressed;
    }

    public enum Device {
        /**
         * A key on the keyboard
         */
        KEYBOARD,

        /**
         * A mouse button
         */
        MOUSE
    }

    /**
     * Gets the button name for the GUI
     *
     * @return The button name
     */
    public String getButtonName() {
        return device.equals(Device.KEYBOARD) ? Keyboard.getKeyName(buttonCode) : Mouse.getButtonName(buttonCode);
    }

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/backend/setting/Bind.kt
package wtf.monsoon.backend.setting

import org.lwjgl.input.Keyboard
import org.lwjgl.input.Mouse

/**
 * @author surge
 * @since 09/02/2023
 */
class Bind(var code: Int, var device: Device) {

    fun isPressed(): Boolean = if (code == 0) { false } else Keyboard.isKeyDown(code) && device == Device.KEYBOARD || Mouse.isButtonDown(code) && device == Device.MOUSE
    fun getButtonName(): String = if (device == Device.KEYBOARD) Keyboard.getKeyName(code) else Mouse.getButtonName(code)

    enum class Device {
        KEYBOARD,
        MOUSE
    }

}